src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -307,8 +340,9 @@ public void paint(final Graphics g, final JComponent c) {
         // performs icon and text rect calculations
         final String text;
         final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
-        if (v != null) {
-            // use zero insets for view since layout only handles text calculations
+        if (v != null && b.getIcon() == null) {
+            // use zero insets for view when HTML contains an image
```

Comment:
I guess it should be "**does not** contain an image", right?

Reply:
This comment is confusing now that it deals with button icon images and HTML images. I updated the comment